### PR TITLE
Export markdown

### PR DESCRIPTION
**Pull request by RCmerci**

### CODE REVIEW

**Comment by tiensonqin** [comment]
@RCmerci To make the output markdown more similar to the standards, we might need to add two options:
1. convert headings to unordered lists
2. ignore properties
What do you think? We can merge this PR now and add the options in a following PR.

**Comment by RCmerci** [comment]
> convert headings to unordered lists

I'll implement this after #27 get merged ('unordered' option required in that PR).

> ignore properties

yes, properties are ignored in this PR.  `lib/export/markdown.ml:269`

**Comment by tiensonqin** [comment]
> And the properties syntax is far from the standards, option for it is unnecessary?

Agreed!

> I'll implement this after #27 get merged ('unordered' option required in that PR).

We need to provide users a way to migrate the old files to the new unordered lists as bullets, the app will request mldoc to convert existing headings to unordered lists for later usage.

File: lib/export/markdown.ml [hunk 1]
```diff
@@ -0,0 +1,550 @@
...
+type refs = Reference.parsed_t
+
+let src_block ?(lang = None) ?(options = None) typ sl =
+  let options =
+    match options with
+    | None -> ""
+    | Some v -> String.concat " " v
+  in
+  List.flatten
+    [ [ raw_text @@ "#+BEGIN_" ^ typ
```
[the remark this answers is not in the document]
Comment (RCmerci) [reply lib/export/markdown.ml]: removed `src_block` . And ignore Export, CommentBlock, Custom ast now

File: lib/export/markdown.ml [hunk 2]
```diff
@@ -0,0 +1,550 @@
...
+and latex_fragment = function
+  | Inline s -> [ Space; raw_text "$"; raw_text s; raw_text "$"; Space ]
+  | Displayed s -> [ Space; raw_text "$$"; raw_text s; raw_text "$$"; Space ]
+
+and macro refs state config m =
+  if m.name <> "embed" then
+    map_raw_text
+    @@
+    if List.length m.arguments > 0 then
+      [ "{{{"; m.name; "("; String.concat "," m.arguments; ")}}}" ]
```
[the remark this answers is not in the document]
Comment (RCmerci) [reply lib/export/markdown.ml]: fixed

File: lib/export/markdown.ml [hunk 3]
```diff
@@ -0,0 +1,550 @@
...
+    [ [ raw_text "```"; Space; raw_text @@ Option.default "" language; Newline ]
+    ; map_raw_text lines
+    ; [ raw_text "```"; Newline ]
+    ]
+
+and quote refs state config tl =
+  flatten_map
+    (fun l ->
+      List.flatten
+        [ [ RawText ">"; Space ]; block refs state config l; [ Newline ] ])
```
[the remark this answers is not in the document]
Comment (RCmerci) [reply lib/export/markdown.ml:353]: Yes, for example:
```
###
 > `asd` ddddd
 > jdiwdjw


 > jidjwid
``` 
is exported as 
```
###
> `asd` ddddd
jdiwdjw


> jidjwid


```

**Comment by RCmerci** [comment]
@tiensonqin 'heading-to-list' option is added now at mldoc, and related commit here https://github.com/logseq/logseq/pull/1444/commits/9ac04980bd98a935c0e4bb4c2c0ce276317b9406 at logseq

File: lib/export/markdown.ml [hunk 4]
```diff
@@ -0,0 +1,550 @@
...
+  | Displayed s -> [ Space; raw_text "$$"; raw_text s; raw_text "$$"; Space ]
+
+and macro refs state config m =
+  if m.name <> "embed" then
+    map_raw_text
+    @@
+    if List.length m.arguments > 0 then
+      [ "{{{"; m.name; "("; String.concat "," m.arguments; ")}}}" ]
+    else
+      [ "{{{"; m.name; "}}}" ]
```
[the remark this answers is not in the document]
Comment (RCmerci) [reply lib/export/markdown.ml]: fixed

File: lib/export/markdown.ml [hunk 5]
```diff
@@ -0,0 +1,550 @@
...
+  else
+    macro_embed refs state config m
+
+and macro_embed refs state config { arguments; _ } =
+  if List.length arguments <> 1 then
+    []
+  else
+    let arg = String.trim (List.hd arguments) in
+    let value = String.(trim @@ sub arg 2 (length arg - 4)) in
+    let raw_result = map_raw_text [ "{{{embed "; arg; "}}}" ] in
```
[the remark this answers is not in the document]
Comment (RCmerci) [reply lib/export/markdown.ml]: fixed

**Comment by tiensonqin** [comment]
@RCmerci awesome!

**Pull request merged**